pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/SegmentDeletionManager.java

```diff
@@ -193,4 +211,29 @@ protected void removeSegmentFromStore(String tableName, String segmentId) {
       LOGGER.info("localDiskDir is not configured, won't delete segment {} from disk", segmentId);
     }
   }
+
+  protected void removeSegmentsFromDeletedDirectory(int deleteRetention) {
```

Comment:
@mayankshriv Identifying what files need to be deleted takes 2 lines of code and performing deletion is ~3 lines of code. `removeSegmentsFromDeletedDirectory()` also performs the deletion of empty directories which has the similar logic (identifying and deletion). Then, this function (~20 lines) needs to be separated into 4 short functions. I feel maybe we are over-splitting the function. What would be the better approach?